### ISSUE TITLE
Add the possibility to download files and keep them in a variable

### ISSUE THREAD

**Issue by TytoCapensis** [TheHive-Project/TheHive4py#322]
As today, all functions related to download of files take a file path as an argument, and download themselves the file on the filesystem (by using the functions in the `session.py` module)

This can cause issues because:
- Downloading a file can be cumbersome in some contexts, sometimes it is preferable to keep it in memory (i.e. in a variable)
- We do not have control about how the file is opened and written exactly

I intend to add a pull request that:
- Allows to pass `False` as a "filepath" to all file download functions, to tell that we do not want the file to be written on the filesystem, but instead returned by the function so we can store it as a variable
- Assume `False` as a default for all file download functions (currently there is no default, the user has to provide a file path regardless).

**Comment by TytoCapensis** [comment]
I just pushed a replacement commit that went with the method you suggested instead (using a `io.BytesIO` object passed in the argument). I still have to modify some files however, to change the type hinting definitions.

Note that I had to use a `# type: ignore` in `session.py`, because mypy got confused and was reporting that `io.BytesIO` was not a correct type for `open()` function (yet it is in another `if` block). If you have a better idea to go around that, I will change the code.
